File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
using namespace std;

bool is_prime(int n)
{
    if (n == 1)
        return false;
    for (int i = 2; i * i <= n; i++){
        if (n % i == 0)
            return false;
    }
    return true;
}

int solution(vector<int> nums) {
    int answer = 0;
    for (int i = 0; i < nums.size(); i++){
        for (int j = i + 1; j < nums.size(); j++){
            for (int k = j + 1; k < nums.size(); k++){
                if (is_prime(nums[i] + nums[j] + nums[k])){
                    answer++;
                }
            }
        }
    }
    return answer;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
import math

def solution(nums):
    answer = 0
    number_len = len(nums)
    for i in range(number_len):
        for j in range(i + 1, number_len):
            for k in range(j + 1, number_len):
                if is_prime(nums[i] + nums[j] + nums[k]):
                    answer += 1
    return answer

def is_prime(n):
    if n == 1:
        return False
    for i in range(2, math.floor(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True
